Change custom page files to use a `.md` extension.

zedseven opened the pull request:
My rationale for this PR is in #320. Closes #320.

To summarise, the lack of a `.md` extension causes problems for external tools working with the custom page files, and my solution is to simply add the extension to the end.

e.g.
- `{name}.page.md`
- `{name}.patch.md`

This *will* be a breaking change for anyone using custom pages.

niklasmohrin commented:
As I outlined in the issue thread, let's first find out how we want to handle this breaking change if we want it. Happy to continue discussion here once we are all on the same page :)

dbrgn commented:
As commented in https://github.com/dbrgn/tealdeer/issues/320#issuecomment-1937775105, I think it's valuable to print a warning when pages with `.page` or `.patch` extension are found in the custom pages directory. This will be a small performance hit, but we can remove it again after 1-2 releases.

@zedseven would you be willing to add such a check?

dbrgn commented:
Could you add a breaking change warning to the docs?

```diff
diff --git a/docs/src/usage_custom_pages.md b/docs/src/usage_custom_pages.md
index 7024abd..d5d0f89 100644
--- a/docs/src/usage_custom_pages.md
+++ b/docs/src/usage_custom_pages.md
@@ -1,5 +1,13 @@
 # Custom Pages and Patches

+> ⚠️ **Breaking change in version 1.7.0:** The file name extension for custom
+> pages and patches was changed:
+>
+> - `<name>.page` → `<name>.page.md`
+> - `<name>.patch` → `<name>.patch.md`
+>
+> If you have custom pages or patches, you need to rename them.
+
 Tealdeer allows creating new custom pages, overriding existing pages as well as
 extending existing pages.
```

zedseven commented:
I've added the temporary check (when looking up any command), added your warning to the docs, rebased the branch onto the latest `main`, and fixed a bug I noticed with my initial implementation.

dbrgn commented on src/cache.rs:
```diff
@@ -407,6 +409,32 @@ impl Cache {
+                } else {
+                    false
+                }
+            });
```
This check could have been written without the `WalkDir` crate, just using `fs::read_dir`. But I started rewriting it, and noticed that the `WalkDir` API is clearly nicer, so let's stick to it 🙂

dbrgn commented on src/cache.rs:
```diff
@@ -395,6 +413,39 @@ impl Cache {
+                }
+            });
+        if old_custom_pages_exist {
+            print_warning(
```
I changed this code to use `print_warning`.

dbrgn reviewed (approved):
Great, thanks! I pushed some minor tweaks.
